base/genericmemory.jl

```diff
@@ -25,8 +37,15 @@ Memory
 """
     AtomicMemory{T} == GenericMemory{:atomic, T, Core.CPU}
 
-One-dimensional dense array with elements of type `T`, where each element is
-independently atomic when accessed, and cannot be set non-atomically.
+!!! warning
```

Comment:
The markdown here is broken because of the indentation, note the empty admonition:

![AtomicMemory](https://github.com/JuliaLang/julia/assets/4944410/bb13e121-9aaa-46b8-b4a8-f700bf7bda59)

Also, I'm not sure how nice it is for a doc string to immediately start with a warning admonition, perhaps just put the admonition after the first paragraph:

```
Fixed-size [`DenseVector{T}`](@ref DenseVector). Each element is independently atomic when accessed, and cannot be set non-atomically.

!!! warning
    Currently the `@atomic` macro and higher level interface have not been completed,
```

Reply:
I'm not sure if "each element is independently atomic when accessed, and cannot be set non-atomically." is too vague. The precise meaning is that 
* `x = mem[i]` will be performed atomically (with `:monotonic` ordering)
* to store `x` somewhere in `mem` the use of `@atomic` (+ specifying ordering?) macro is necessary, i.e. `@atomic :sequentially_consistent mem[i] = x` is the only legal way to do so (without dropping to `Core.memoryref[...]` buildins).  

How about 
```
Fixed-size [`DenseVector{T}`](@ref DenseVector). Access to its any of its elements is performed atomically (with `:monotonic` ordering). Moreover setting any of the elements must be accomplished using the `@atomic` macro and explicitly specifying ordering.
```